packages/core/src/application.ts

```diff
@@ -76,19 +79,44 @@ export class Application extends Context {
    * use the app.server function instead.
    * ```ts
    * app.servers([
-   *  new RestServer(app, options),
-   *  new GRPCServer(app, options),
+   *  RestServer,
+   *  GRPCServer,
    * ]);
    * // Creates a binding for "servers.RestServer" and a binding for
    * // "servers.GRPCServer";
    * ```
    *
-   * @param {Array<Server>} servers An array of instantiated Server instances.
-   * @returns {Binding}
+   * @param {Constructor<Server>[]} ctors An array of Server constructors.
+   * @memberof Application
+   */
+  public servers<T extends Server>(ctors: Constructor<T>[]) {
+    ctors.map(ctor => this.server(ctor));
+  }
+
+  /**
+   * Retrieve the singleton instance for a bound constructor.
+   *
+   * @template T
+   * @param {Constructor<T>} ctor The constructor that was used to make the
+   * binding.
+   * @returns {Promise<T>}
+   * @memberof Application
+   */
+  public async getServer<T extends Server>(ctor: Constructor<T>): Promise<T> {
+    return await this.get(`servers.${ctor.name}`) as T;
+  }
+
+  /**
+   * Retrieve the singleton instance for a bound constructor synchronously.
+   * Whenever possible, you should make use of the async "getServer" instead.
+   *
+   * @template T
+   * @param {Constructor<T>} ctor
+   * @returns {T}
    * @memberof Application
    */
-  public servers(servers: Server[]): Binding[] {
-    return servers.map(server => this.server(server));
+  public getServerSync<T extends Server>(ctor: Constructor<T>): T {
```

Comment:
Considering that this is used primarily in tests, and that async functions are cheap to write, is this sync helper necessary at all?